### ISSUE TITLE
Find neighboring nodes.

### ISSUE THREAD

**Issue by suziyou12** [agoodge/LUNAR#1]
Why are the neighbors of all data points found within the training data when constructing the graph?

**Comment by agoodge** [comment]
Hi @suziyou12 

We find nearest neighbors from only the training set because they are all normal points, instead of the test set where some points are anomalies.
